eva-pipeline/src/test/java/embl/ebi/variation/eva/pipeline/steps/VariantsAnnotLoadBatchTest.java

```diff
@@ -0,0 +1,195 @@
+        int count = 0;
+        while ((variantAnnotation = reader.read()) != null) {
+            count++;
+            if (variantAnnotation.getConsequenceTypes() != null && !variantAnnotation.getConsequenceTypes().isEmpty()) {
+                consequenceTypeCount++;
+            }
+        }
+        // all should have at least consequence type annotations
+        assertEquals(count, consequenceTypeCount);
+
+        // variantAnnotationReader should get all the lines from the file
+        long actualCount = JobTestUtils.getLines(new GZIPInputStream(new FileInputStream(vepOutput)));
+        assertEquals(actualCount, count);
+
+    }
+
+    @Test
+    public void variantAnnotationWriterShouldWriteAllFieldsIntoMongoDb() throws Exception {
+        String dbName = "VariantAnnotLoadBatchTest_writer";
+        JobTestUtils.cleanDBs(dbName);
+
+        // first do a mock of a "variants" collection, with just the _id
+        VariantAnnotationLineMapper lineMapper = new VariantAnnotationLineMapper();
+        List<VariantAnnotation> annotations = new ArrayList<>();
+        for (String annotLine : vepOutput.split("\n")) {
+            annotations.add(lineMapper.mapLine(annotLine, 0));
+        }
+        MongoClient mongoClient = new MongoClient();
+        DBCollection variants = mongoClient.getDB(dbName).getCollection("variants");
+        DBObjectToVariantAnnotationConverter converter = new DBObjectToVariantAnnotationConverter();
+        for (VariantAnnotation annotation : annotations) {
+            String id = buildStorageId(annotation.getChromosome(), annotation.getStart(),
+                    annotation.getReferenceAllele(), annotation.getAlternativeAllele());
+            try {
+                variants.insert(new BasicDBObject("_id", id));
+            } catch (MongoException.DuplicateKey e) {
+                ; // ignore, we are just scaffolding the variants collection
+            }
+        }
+
+        // now, load the annotation
+        MongoOperations mongoOperations = new MongoTemplate(new MongoClient(), dbName);
+        MongoItemWriter<VariantAnnotation> writer = variantsAnnotLoadBatch.initWriter("variants", mongoOperations);
+
+        writer.write(annotations);
+
+        // and finally check that documents in DB have annotation (only consequence type)
+        DBCursor cursor = variants.find();
+
+        int cnt=0;
+        int consequenceTypeCount = 0;
+        while (cursor.hasNext()) {
+            cnt++;
+            VariantAnnotation annot = converter.convertToDataModelType((DBObject)cursor.next().get("annot"));
+            assertTrue(annot.getConsequenceTypes() != null);
+            consequenceTypeCount += annot.getConsequenceTypes().size();
+        }
+        assertTrue(cnt>0);
+        assertEquals(annotations.size(), consequenceTypeCount);
+
+        JobTestUtils.cleanDBs(dbName);
+    }
+
+    private String buildStorageId(String chromosome, int start, String reference, String alternate) {
```

Comment:
We could even use the Mongo?ConnectionHelper since this is a pretty much Mongo-specific thing.